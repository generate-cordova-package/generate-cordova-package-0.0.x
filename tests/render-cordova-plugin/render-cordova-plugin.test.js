/* global it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const renderCordovaPluginTree = require('../../lib/cordova-plugin/render')

const renderCordovaPluginProperties = require('./render-cordova-plugin-properties.config')

it('renders fs-tree object correctly for cordova-plugin', () => {
  const outputTree = renderCordovaPluginTree(renderCordovaPluginProperties)

  expect(outputTree).toMatchSnapshot()
})
