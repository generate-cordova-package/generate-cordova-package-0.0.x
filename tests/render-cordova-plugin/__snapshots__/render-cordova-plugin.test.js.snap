// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders fs-tree object correctly for cordova-plugin 1`] = `
Object {
  "test-plugin": Object {
    "README.md": "# test-plugin

Copyright 2023-present Alice <alice@alice.com>
",
    "package.json": "{
  \\"name\\": \\"test-plugin\\",
  \\"version\\": \\"0.1.1\\",
  \\"// copyright\\": \\"Copyright 2023-present Alice <alice@alice.com>\\",
  \\"description\\": \\"test plugin\\",
  \\"license\\": \\"ISC\\",
  \\"author\\": {
    \\"name\\": \\"Alice\\",
    \\"email\\": \\"alice@alice.com\\"
  }
}
",
    "plugin.xml": "<plugin id=\\"cc.testplugin\\" version=\\"0.1.1\\">
  <js-module src=\\"www/testModule.js\\" name=\\"testObject\\">
    <runs/>
  </js-module>
</plugin>
",
    "www": Object {
      "testModule.js": "// Copyright 2023-present Alice <alice@alice.com>

function testFunction (callback) {
  callback('pong')
}

window.testObject = {
  testFunction: testFunction
}
",
    },
  },
}
`;
